lib/jgi_mg_assembly/pipeline_steps/agp.py

```diff
@@ -0,0 +1,60 @@
+from step import Step
+import os
+from jgi_mg_assembly.utils.util import mkdir
+
+
+AGP_FILE_TOOL = "/kb/module/bbmap/fungalrelease.sh"
+class AgpRunner(Step):
+    def __init__(self, scratch_dir, output_dir):
+        super(AgpRunner, self).__init__("createAGPFile", "BBTools", AGP_FILE_TOOL, scratch_dir, output_dir, False)
+
+    def run(self, spades_scaffolds):
+        """
+        Runs bbmap/fungalrelease.sh to build AGP files and do some mapping and cleanup.
+        Returns a dictionary where values are paths to files and keys are the following:
+        scaffolds - mapped scaffolds fasta file
+        contigs - mapped contigs fasta file
+        agp - AGP file
+        legend - legend for generated scaffolds
+        """
+        if spades_scaffolds is None or not os.path.exists(spades_scaffolds):
+            has_contigs = os.path.exists(os.path.join(spades_dir, "contigs.fasta"))
```

Comment:
I don't see `spades_dir` defined anywhere

Reply:
You're right. Tweaked to get the spades contigs file directly.